source/_integrations/tesla_fleet.markdown

```diff
@@ -36,178 +36,295 @@ ha_platforms:
-2. Navigate to the [Developer dashboard](https://developer.tesla.com/en_US/dashboard). Change your locale if needed to wherever your account is based, using the globe icon at the top-right corner.
-3. Select **Create New Application**. This should launch a new page with the header **Create Fleet API Application**.
-4. For simple integrations, under **Registration Type**, select **Just for me**. If you're confident about your situation being a business setup instead, select **For my business**. Select **Next**.
-5. At the **Application Details** step, use a name that is easy to refer to later, such as `ha-integration`. This will be needed later while configuring the integration. Select **Next**.
-6. At the **Client Details** step, under **Oauth Grant Type**, select **Authorization Code and Machine-to-Machine**.
-   - Under **Allowed Origin URL(s)**, enter your domain where you hosted your public key earlier. Using the example above, this would be `https://my.domain.com/`.
-   - Under **Allowed Redirect URI(s)**, enter `https://my.home-assistant.io/redirect/oauth`.
-   - Select **Next**.
-7. At the **API & Scopes** step, select the configurations you'd like to access.
-   - At least one of `Vehicle Information` or `Energy Product Information` **must** be selected for the integration to function. It is recommended you select all scopes for full functionality.
-   - If you're unsure, you can select only one of these two required scopes for now and update the scopes later from the Developer Dashboard. However, note that if the scopes are updated, you will need to reconfigure the integration fully (refer to the **Integration is broken and needs to be reconfigured** troubleshooting steps below).
-   - Select **Next**.
-8. At the **Billing Details (Optional)** step, enter your billing details if needed, then select **Submit**. Tesla does not support billing in all countries yet. **Developers in countries that do not yet support payments will not see this step**.
-   - Tesla provides a $10 monthly credit for personal API usage, so your level of usage may be covered for free. Detailed pricing info for commands, data polling, and wake signals can be found at [developer.tesla.com](https://developer.tesla.com). Use these details to calculate your usage estimate if you rely heavily on this integration.
-   - If unsure, you can select **Skip & Submit** for now and add the billing details later when your usage is close to the free threshold.
-9. Your developer application is ready for use!
+{% details "Using external web hosting" %}
+
+Use this method if you have your own web hosting or prefer using a third-party service.
+
+1. **Open a terminal** on your computer.
+
+2. **Create your private key**:
+
+   ```shell
+   openssl ecparam -name prime256v1 -genkey -noout -out tesla_fleet.key
+   ```
+
+3. **Create the public key**:
+
+   ```shell
+   openssl ec -in tesla_fleet.key -pubout -out public-key.pem
+   ```
+
+4. **Rename the public key** file:
```

Comment:
```suggestion
1. Open a terminal on your computer.

2. Create your private key:

   ```shell
   openssl ecparam -name prime256v1 -genkey -noout -out tesla_fleet.key
   ```

3. Create the public key:

   ```shell
   openssl ec -in tesla_fleet.key -pubout -out public-key.pem
   ```

4. Rename the public key file:
```

Reply:
I think the suggestion broke so I did this manually.